### PR TITLE
feat: creates a wallet command to create a signature and a nonce from a private key and challenge 

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
It is desirable to have a wallet cli command to create and display (on terminal) a signature and nonce, from the given of a private key and challenge.

Motivation and Context
---
This PR addresses the above task.

How Has This Been Tested?
---
Unit tests

### CODE REVIEW

File: applications/tari_console_wallet/src/automation/commands.rs [hunk 1]
```diff
@@ -652,6 +685,44 @@ pub async fn command_runner(
...
+                },
+                Err(e) => eprintln!("CreateAggregateSignatureUtxo error! {}", e),
+            },
+            SignMessage(args) => match sign_message(args.private_key, args.challenge) {
+                Ok((sgn, nonce)) => {
+                    println!(
+                        "Sign message: 
+                                1. signature: {},
+                                2. public key: {}",
+                        *Zeroizing::new(sgn.get_signature().to_hex()),
```
Comment (SWvheerden) [inline applications/tari_console_wallet/src/automation/commands.rs]: We dont have to Zeroize this, the signature is public anyway. 
The argument private key is not thou.

File: applications/tari_console_wallet/src/automation/commands.rs [hunk 2]
```diff
@@ -253,6 +275,17 @@ pub async fn coin_split(
...
 
+pub fn sign_message(private_key: String, challenge: String) -> Result<(Signature, RistrettoSecretKey), CommandError> {
+    let private_key =
+        PrivateKey::from_hex(private_key.as_str()).map_err(|e| CommandError::InvalidArgument(e.to_string()))?;
+    let challenge = challenge.as_bytes();
+
+    let nonce = PrivateKey::random(&mut OsRng);
+    let signature = Signature::sign(private_key, nonce.clone(), challenge).map_err(CommandError::FailedSignature)?;
+
+    Ok((signature, nonce))
```
Comment (SWvheerden) [inline applications/tari_console_wallet/src/automation/commands.rs]: We need to output the public nonce here, not the private one. 
Signatures are:
`sig = r + e*k`
and you publish: `sig and R`, which is the public version of `r`

File: applications/tari_console_wallet/src/automation/commands.rs (hunk 1 above)
Comment (jorgeantonio21) [reply applications/tari_console_wallet/src/automation/commands.rs]: Resolved !

File: applications/tari_console_wallet/src/automation/commands.rs (hunk 2 above)
Comment (jorgeantonio21) [reply applications/tari_console_wallet/src/automation/commands.rs]: Thanks for the explanation ! I have refactored the code accordingly .

File: applications/tari_console_wallet/src/automation/commands.rs [hunk 3]
```diff
@@ -272,6 +275,18 @@ pub async fn coin_split(
     Ok(tx_id)
 }
 
+pub fn sign_message(private_key: String, challenge: String) -> Result<(Signature, PublicKey), CommandError> {
```
Comment (SWvheerden) [inline applications/tari_console_wallet/src/automation/commands.rs]:
```suggestion
pub fn sign_message(private_key: String, challenge: String) -> Result<Signature, CommandError> {
```

File: applications/tari_console_wallet/src/automation/commands.rs [hunk 4]
```diff
@@ -272,6 +275,18 @@ pub async fn coin_split(
...
+    let private_key =
+        PrivateKey::from_hex(private_key.as_str()).map_err(|e| CommandError::InvalidArgument(e.to_string()))?;
+    let challenge = challenge.as_bytes();
+
+    let nonce = PrivateKey::random(&mut OsRng);
+    let signature = Signature::sign(private_key, nonce.clone(), challenge).map_err(CommandError::FailedSignature)?;
+
+    let public_nonce = PublicKey::from_secret_key(&nonce);
+    Ok((signature, public_nonce))
```
Comment (SWvheerden) [inline applications/tari_console_wallet/src/automation/commands.rs]:
```suggestion
    Ok(signature)
```

File: applications/tari_console_wallet/src/automation/commands.rs [hunk 5]
```diff
@@ -697,6 +712,18 @@ pub async fn command_runner(
                 },
                 Err(e) => eprintln!("CreateAggregateSignatureUtxo error! {}", e),
             },
+            SignMessage(args) => match sign_message(args.private_key, args.challenge) {
+                Ok((sgn, nonce)) => {
```
Comment (SWvheerden) [inline applications/tari_console_wallet/src/automation/commands.rs]:
```suggestion
                Ok(sgn) => {
```

File: applications/tari_console_wallet/src/automation/commands.rs [hunk 6]
```diff
@@ -697,6 +712,18 @@ pub async fn command_runner(
...
+                    println!(
+                        "Sign message: 
+                                1. signature: {},
+                                2. public key: {}",
+                        sgn.get_signature().to_hex(),
+                        nonce.to_hex(),
```
Comment (SWvheerden) [inline applications/tari_console_wallet/src/automation/commands.rs]:
```suggestion
                        sgn.get_public_nonce().to_hex(),
```

File: applications/tari_console_wallet/src/automation/commands.rs [hunk 7]
```diff
@@ -272,6 +275,18 @@ pub async fn coin_split(
...
 
+pub fn sign_message(private_key: String, challenge: String) -> Result<(Signature, PublicKey), CommandError> {
+    let private_key =
+        PrivateKey::from_hex(private_key.as_str()).map_err(|e| CommandError::InvalidArgument(e.to_string()))?;
+    let challenge = challenge.as_bytes();
+
+    let nonce = PrivateKey::random(&mut OsRng);
+    let signature = Signature::sign(private_key, nonce.clone(), challenge).map_err(CommandError::FailedSignature)?;
+
+    let public_nonce = PublicKey::from_secret_key(&nonce);
```
Comment (SWvheerden) [inline applications/tari_console_wallet/src/automation/commands.rs]:
```suggestion
```

**Review by SWvheerden — COMMENTED** [review]
Looks good,
The signature already contains the public nonce inside of it.

**Pull request merged**